src/middlewared/middlewared/alembic/versions/21.MM/2021-05-25_22-00_afp_to_smb.py

```diff
@@ -0,0 +1,167 @@
+Create Date: 2021-05-25 19:00:34.776266+00:00
+
+"""
+import json
+import textwrap
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '5f4cab067712'
+down_revision = 'bd637e18fb0b'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('cifs_timemachine_quota', sa.Integer(), nullable=True))
+
+    op.execute("UPDATE sharing_cifs_share SET cifs_timemachine_quota = 0")
+
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.alter_column('cifs_timemachine_quota', existing_type=sa.INTEGER(), nullable=False)
+
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.add_column(sa.Column('cifs_afp', sa.Boolean(), nullable=True))
+
+    op.execute("UPDATE sharing_cifs_share SET cifs_afp = 0")
+    op.execute("""
+        UPDATE sharing_cifs_share SET cifs_purpose = 'NO_PRESET', cifs_afp = 1 WHERE cifs_purpose = 'MULTI_PROTOCOL_AFP'
+    """)
+
+    with op.batch_alter_table('sharing_cifs_share', schema=None) as batch_op:
+        batch_op.alter_column('cifs_afp', existing_type=sa.BOOLEAN(), nullable=False)
+
+    conn = op.get_bind()
+    has_cifs_home = bool(conn.execute("SELECT * FROM sharing_cifs_share WHERE cifs_home = 1"))
+    disable_acl_if_trivial = []
+    for share in conn.execute("SELECT * FROM sharing_afp_share").fetchall():
+        cifs_auxsmbconf = []
+        share_disable_acl_if_trivial = False
+        if share["afp_allow"].strip():
+            cifs_auxsmbconf.append(f"valid users = {share['afp_allow'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_deny"].strip():
+            cifs_auxsmbconf.append(f"invalid users = {share['afp_deny'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_ro"].strip():
+            cifs_auxsmbconf.append(f"rolist = {share['afp_ro'].strip()}")
+            share_disable_acl_if_trivial = True
+        if share["afp_rw"].strip():
+            cifs_auxsmbconf.append(f"rwlist = {share['afp_rw'].strip()}")
+            share_disable_acl_if_trivial = True
+        umask_inv = "%o" % (~int(share["afp_umask"], 8) & 0o777)
+        cifs_auxsmbconf.append(f"create mask = 0{umask_inv}")
+        cifs_auxsmbconf.append(f"directory mode = 0{umask_inv}")
+        cifs_auxsmbconf.append(f"force create mode = 0{share['afp_fperm']}")
+        cifs_auxsmbconf.append(f"force directory mode = 0{share['afp_dperm']}")
+        if share["afp_auxparams"].strip():
+            cifs_auxsmbconf.append(textwrap.indent(f"\nNetatalk Auxiliary Parameters:\n\n{share['afp_auxparams']}", "; "))
+
+        cifs_share = {
+            "cifs_purpose": "NO_PRESET",
+            "cifs_path": share["afp_path"],
+            "cifs_path_suffix": "",
```

Comment:
We should set `%U` as a path suffix if `has_cifs_home` and `share["afp_home"]`

This is to approximately preserve behavior of `basedir regex` from afp.conf.